### PR TITLE
Added support for No Version Warning

### PR DESCRIPTION
**Pull request by emipa606**
Since I got requests of adding support for my [No Version Warning-mod](https://steamcommunity.com/sharedfiles/filedetails/?id=2599504692) to RimPy but never could since that one is closed source, so Ill try here instead.

If the mod is subscribed it adds the supported version-tags to the applicable mods when generating the cache.
Can probably be improved and/or put behind some option but I'm not that experienced with GUI-coding in Python to do that.
Hopefully you will get any use of this.

Nice initiative with the project!

### CODE REVIEW

File: view/main_content_panel.py [hunk 1]
```diff
@@ -262,11 +268,60 @@ def refresh_cache_calculations(self) -> None:
...
             f"Combined {len(self.expansions)} expansions, {len(self.local_mods)} local mods, and {len(self.workshop_mods)}. Total elements to get dependencies for: {len(all_mods)}"
         )
 
+        # Check mods verified working with newer game-versions than defined
+        # https://steamcommunity.com/sharedfiles/filedetails/?id=2599504692
+        package_id_to_uuid = {}
+        for mod_uuid, modmetadata in all_mods.items():
+            package_id_to_uuid[modmetadata["packageId"]] = mod_uuid
+        if "mlie.noversionwarning" in package_id_to_uuid.keys():
+            no_version_mod = all_mods[package_id_to_uuid["mlie.noversionwarning"]]
```
Comment (oceancabbage) [inline view/main_content_panel.py:283]: Minor: may want to extract this to a function in the module

**Comment by oceancabbage** [comment]
This looks really good. Was able to test on my machine that this works as expected. Only left one minor comment but I'm happy with merging this.

**Review by twstagg — CHANGES_REQUESTED** [review]
I like the idea. I agree with @oceancabbage previous comment. It is of my opinion that:

    This feature should be made configurable (default: off) so that users can opt out of using it entirely.

File: view/main_content_panel.py (hunk 1 above)
Comment (emipa606) [reply view/main_content_panel.py:283]: That code was cloned from another part of the mod so it would probably be a good idea

**Comment by twstagg** [comment]
please cherry pick your commits and rebase off of main again - sprint-a1.0.3 has been merged

**Pull request closed without merging**